spyder/widgets/findreplace.py

```diff
@@ -444,10 +461,41 @@ def replace_find(self, focus_replace_text=False):
                                        QTextCursor.KeepAnchor)
                 else:
                     break
-                if not self.all_check.isChecked():
+                if not replace_all:
                     break
-            self.all_check.setCheckState(Qt.Unchecked)
             if cursor is not None:
                 cursor.endEditBlock()
             if focus_replace_text:
                 self.replace_text.setFocus()
+
+    @Slot()
+    def replace_find_all(self, focus_replace_text=False):
+        """Replace and find all matching occurrences"""
+        self.replace_find(focus_replace_text, replace_all=True)
+
+                
+    @Slot()
+    def replace_find_selection(self, focus_replace_text=False):

```

Comment:
Actually a bigger inconsistency is that I didn't implement the whole word restriction. DOH! And it might ultimately be simpler to use the QT implementation because python's re doesn't handle whole words. Let me give this some thought over the weekend.